Firebase/Core/FIRApp.m

```diff
@@ -796,4 +806,31 @@ - (void)sendLogsWithServiceName:(NSString *)serviceName
 }
 #pragma clang diagnostic pop
 
+#pragma mark - App Life Cycle
+
+- (void)subscribeForAppDidBecomeActiveNotifications {
+#if TARGET_OS_IOS || TARGET_OS_TV
+  NSNotificationName notificationName = UIApplicationDidBecomeActiveNotification;
+#endif
+
+#if TARGET_OS_OSX
+  NSNotificationName notificationName = NSApplicationDidBecomeActiveNotification;
+#endif
+
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(appDidBecomeActive:)
+                                               name:notificationName
+                                             object:nil];
+}
+
+- (void)appDidBecomeActive:(NSNotification *)notification {
+  [self logDiagnostics];
+}
+
+- (void)logDiagnostics {
+  if ([self isDataCollectionDefaultEnabled]) {
+    [FIRCoreDiagnosticsConnector logConfigureCoreWithOptions:_options];
```

Comment:
Did you want to create a new logging method, or keep as-is?

Reply:
I want to log the same data. Now I think, `logConfigureCoreWithOptions` can be renamed to something like `logCoreDataWithOptions`. WDYT?